src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/PipeListener.cs

```diff
@@ -19,13 +19,15 @@ public PipeListener(ServiceContext serviceContext) : base(serviceContext)
         /// <summary>
         /// Creates the socket used to listen for incoming connections
         /// </summary>
-        protected override UvStreamHandle CreateListenSocket()
+        protected override void CreateListenSocket()
```

Comment:
Did you consider the following?
```
try
{
    socket.Init(...);
    socket.Bind(...);
    socket.Listen(...);
}
catch
{
    socket.Dispose();
    throw;
}
return socket;
```
I think this would be just as safe, and a smaller change overall.